added movie details activity, created markup for this activity

jeksonshar opened the pull request:
need to create changing the background height depending on the orientation of the phone

jeksonshar committed:
added movie details activity, created markup for this activity

Timurovic commented on app/src/main/AndroidManifest.xml:
```diff
@@ -9,6 +9,7 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/Theme.FunAcademyApp">
+        <activity android:name=".MovieDetailsActivity"></activity>
```
в таком случае можно написать `<activity android:name=".MovieDetailsActivity" />`

Timurovic commented on app/src/main/res/layout/activity_movie_details.xml:
```diff
@@ -0,0 +1,323 @@
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintBottom_toBottomOf="parent"
+            android:contentDescription="@string/background" />
```
рекомендовано для дочерних элементов constraint задавать id

Timurovic commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,4 +1,18 @@
 <resources>
     <string name="app_name" translatable="false">FunAcademyApp</string>
     <string name="hello">Hello world!</string>
+    <string name="back">Back</string>
+    <string name="_13">13+</string>
+    <string name="avengers_end_game">Avengers:<br/>End Game</string>
+    <string name="action_adventure_fantasy">Action, Adventure, Fantasy</string>
+    <string name="background">background</string>
+    <string name="_125_reviews">125 REVIEWS</string>
+    <string name="storyline">Storyline</string>
+    <string name="description_detail_movie">After the devastating events of Avengers: Infinity War, the universe is in ruins. With the help of remaining allies, the Avengers assemble once more in order to reverse Thanos\' actions and restore balance to the universe.</string>
+    <string name="cast">Cast</string>
+    <string name="robert_downey_jr">Robert Downey Jr.</string>
+    <string name="chris_evans">Chris Evans</string>
+    <string name="mark_ruffalo">Mark Ruffalo</string>
+    <string name="chris_hemsworth">Chris Hemsworth</string>
+    <string name="image">image</string>
```
лучше заранее обеспокоится неймингом, с ростом проекта у тебя может появиться дубли по названию, но разные по содержанию
Плюс не всегда ясно, что это. 
Например: back - это кнопка или действие? 
avengers_end_game - скорее заголовок.

jeksonshar replied:
Скажи пожалуйста: ты имеешь ввиду нейминг при определении ID полей в файле разметки? или строковых ресурсов тоже? 
В данном случае мы забиваем текст вручную, и поскольку в домашке было сказано, что нельзя жестко кодировать текст, нужно его в ресурсы, и поскольку каждый этот текст индивидуален именно для этого фильма, то его лучше в параметре name указать аналогично сути, чтобы легче найти. А уже потом, когда для каждого фильма данные будут браться из запроса (я так понял мы это будем делать), эти строки для этой страницы мы просто удалим.
Т.е. по сути это костыль временный.

Timurovic replied:
лучше заранее обеспокоится о нейминге, назови их со смысловыми ключами.
Например можешь добавить название экрана:
`<string name="movie_details_description">`

Timurovic commented on app/src/main/java/com/jeksonshar/funacademyapp/MainActivity.kt:
```diff
@@ -1,11 +1,23 @@
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        val textView: TextView = findViewById(R.id.first_activity_text_view)
+
+        textView.setOnClickListener { moveToNextScreen() }
```
зачем вообще делать переход?
Можно же стартовым экран сделать MovieDetailsActivity

jeksonshar replied:
В домашнем задании, было указано: 
"4. Чтобы открыть MovieDetailsActivity из MainActivity, вы должны выполнить те же шаги, что и мы на 3-м семинаре."
Поэтому так и выполнил. Так все-таки - удалить и сделать одну активити, или сделать согласно 4 пункту ДЗ (т.е. оставить как есть)?

Timurovic replied:
Тогда желательно по кнопке)

jeksonshar committed:
Внесены изменения согласно комментариям в pull request. Остался вопрос - как обрезать imageView, выступающий в роли фона. Добавлены изображение и фон (закомментированы) для замены проблемного фона.

jeksonshar committed:
Изменён переход на страницу MovieDetails, реализован переход по кнопке

Timurovic commented on app/src/main/res/drawable/ic_background.xml:
```diff
@@ -0,0 +1,30 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:aapt="http://schemas.android.com/aapt"
+    android:width="375dp"
+    android:height="812dp"
+    android:viewportWidth="375"
+    android:viewportHeight="812">
+  <path
+      android:pathData="M0,0H375V812H0V0Z"
+      android:fillColor="#191926"/>
+  <group>
+    <clip-path
+        android:pathData="M0,0H375V812H0V0Z"/>
+    <path
+        android:pathData="M-30,0H500V298H-30V0Z"/>
+    <path
+        android:pathData="M0,0H375V298H0V0Z">
+      <aapt:attr name="android:fillColor">
+        <gradient 
+            android:startY="149"
+            android:startX="108.465"
+            android:endY="274.613"
+            android:endX="108.465"
+            android:type="linear">
+          <item android:offset="0" android:color="#02191926"/>
+          <item android:offset="1" android:color="#FF191926"/>
+        </gradient>
+      </aapt:attr>
+    </path>
+  </group>
+</vector>
```
не хватает выравнивания, Ctrl + Shift + L

jeksonshar replied:
Ctrl + Alt + L     у меня сработала эта комбинация

Timurovic commented on app/src/main/res/layout/activity_movie_details.xml:
```diff
@@ -0,0 +1,356 @@
+<!--            app:layout_constraintTop_toTopOf="parent"-->
+<!--            android:contentDescription="@string/background_movie" />-->
+
+        <ImageView
+            android:id="@+id/path_back_view"
+            android:layout_width="6dp"
+            android:layout_height="10dp"
+            android:layout_marginTop="62dp"
+            android:layout_marginStart="18dp"
+            android:src="@drawable/path"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            android:contentDescription="@string/back_text" />
```
маленький элемент, нажать будет тяжело

Timurovic commented on app/src/main/res/layout/activity_movie_details.xml:
```diff
@@ -0,0 +1,356 @@
+<!--            app:layout_constraintTop_toTopOf="parent"-->
+<!--            android:contentDescription="@string/background_movie" />-->
+
+        <ImageView
+            android:id="@+id/path_back_view"
+            android:layout_width="6dp"
+            android:layout_height="10dp"
+            android:layout_marginTop="62dp"
+            android:layout_marginStart="18dp"
+            android:src="@drawable/path"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent"
+            android:contentDescription="@string/back_text" />
+
+        <TextView
+            android:id="@+id/back_view"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="8dp"
+            android:layout_marginTop="56dp"
+            android:text="@string/back_text"
+            android:textColor="@color/gray"
+            app:layout_constraintStart_toEndOf="@+id/path_back_view"
+            app:layout_constraintTop_toTopOf="parent" />
```
лучше объединить как один элемент

Timurovic commented on app/src/main/res/layout/activity_movie_details.xml:
```diff
@@ -0,0 +1,356 @@
+
+        <TextView
+            android:id="@+id/rating_movie"
+            android:layout_width="24dp"
+            android:layout_height="16dp"
```
зачем тут размеры?
желательно не указывать размеры элементов или ограничиться минимальным/максимальным размером

Timurovic commented on app/src/main/res/layout/activity_movie_details.xml:
```diff
@@ -0,0 +1,356 @@
+            android:id="@+id/rating_movie"
+            android:layout_width="24dp"
+            android:layout_height="16dp"
+            android:layout_marginStart="16dp"
+            android:layout_marginTop="226dp"
```
по-хорошему это нужно выносить в dimens

Timurovic commented on app/src/main/res/layout/activity_movie_details.xml:
```diff
@@ -0,0 +1,356 @@
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="16dp"
+            android:layout_marginTop="254dp"
```
тут нужно относительно элементов делать, не относительно родителя

Timurovic commented on app/src/main/res/layout/activity_movie_details.xml:
```diff
@@ -0,0 +1,356 @@
+            android:layout_marginStart="16dp"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintEnd_toStartOf="@id/cast_image_2"
+            app:layout_constraintTop_toBottomOf="@id/cast_movie_view"
```
тут лучше barrier

jeksonshar replied:
барьер где именно? под cast_movie_view? не пойму для чего

Timurovic commented on app/src/main/res/layout/activity_movie_details.xml:
```diff
@@ -0,0 +1,356 @@
+            android:layout_marginStart="8dp"
+            app:layout_constraintStart_toEndOf="@id/cast_image_1"
+            app:layout_constraintEnd_toStartOf="@id/cast_image_3"
+            app:layout_constraintTop_toBottomOf="@id/cast_movie_view"
```
тут лучше barrier

jeksonshar replied:
барьер где именно? под cast_movie_view? не пойму для чего

Timurovic commented on app/src/main/res/layout/activity_movie_details.xml:
```diff
@@ -0,0 +1,356 @@
+            android:layout_marginStart="8dp"
+            app:layout_constraintStart_toEndOf="@id/cast_image_2"
+            app:layout_constraintEnd_toStartOf="@id/cast_image_4"
+            app:layout_constraintTop_toBottomOf="@id/cast_movie_view"
```
тут лучше barrier

jeksonshar replied:
барьер где именно? под cast_movie_view? не пойму для чего

Timurovic commented on app/src/main/res/layout/activity_movie_details.xml:
```diff
@@ -0,0 +1,356 @@
+            android:adjustViewBounds="true"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toEndOf="@id/cast_image_3"
+            app:layout_constraintTop_toBottomOf="@id/cast_movie_view"
```
тут лучше barrier

jeksonshar replied:
барьер где именно? под cast_movie_view? не пойму для чего

Timurovic commented on app/src/main/res/layout/activity_movie_details.xml:
```diff
@@ -0,0 +1,356 @@
+                        image_name_movie_view_4"
+                tools:layout_editor_absoluteY="719dp" />
+
+            <View
+                android:id="@+id/view_indent"
+                android:layout_width="wrap_content"
+                android:layout_height="16dp"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintTop_toBottomOf="@id/barrier_before_indent"/>
```
не совсем понял зачем добавил

jeksonshar replied:
Этот view и barrier перед ним добавил для отступа между последними TextView и низом экрана, чтобы они не лежали прямо на нем, но вижу, что это перебор и тут достаточно было paddingBottom=16 в родителе
уберу их

jeksonshar committed:
Вынесены все размеры и шрифты в файл dimens.xml
Выполнено выравнивание в файле ic_background.xml
Элементы для перехода назад объеденены в LinearLayout для удобства пользователя
Убраны Barrier и View, выполняющие роль отступа снизу страницы

jeksonshar committed:
Установлен фон для NestedScrollView
Установлен ImageView в виде картинки заголовка
Все лишнее убрано
